db/datastore.go

```diff
@@ -0,0 +1,29 @@
+package db
+
+import (
+	"database/sql"
+	"fmt"
+	_ "github.com/lib/pq"
+	"github.com/zzuun/time-tracker/config"
+	"github.com/zzuun/time-tracker/models"
+	"time"
+)
+
+type DataStore interface {
+	Close()
+	CreateUser(models.User) (int, error)
+	GetUser(string) (models.User, error)
+	AddTimeEntry(int) (models.Entry, error)
+	UpdateTimeEntry(int, int) error
+	ListTimeEntries(time.Time, time.Time, int) (string, error)
+}
+
+func NewDataStore() (datastore DataStore, err error) {
```

Comment:
constructor is for the struct ..not for interface.
rename to `func NewDatabase() (datastore DataStore, err error)`